response_operations_ui/views/collection_exercise.py

```diff
@@ -1182,6 +1194,42 @@ def save_ci_versions(short_name: str, period: str, form_type: str):
             ce_details["collection_exercise"]["id"], form_type
         )
         return redirect(url_for("collection_exercise_bp.view_sample_ci_summary", short_name=short_name, period=period))
+    else:
+        ########################################################
+        # WE WON'T DO THIS, IT'S JUST TO TEST THE LOGICAL FLOW #
+        ########################################################
+
+        # We need to re-retrieve the list of available CIR versions (POC exemplar using a Redis cache)
+        redis_cache = RedisCache()
+        survey_ref = redis_cache.get_survey_by_shortname(short_name).get("surveyRef")
```

Comment:
During the subsequent `POST`, also call the cache instead of the explicit controller